File: 0414-third-maximum-number/0414-third-maximum-number.cpp
class Solution {
public:
    int thirdMax(vector<int>& arr) {
        long long largest=LLONG_MIN;
        long long s_largest=LLONG_MIN;
        long long t_largest=LLONG_MIN;
        for(int i=0;i<arr.size();i++){
            if(arr[i]>largest){
                t_largest=s_largest;
                s_largest=largest;
                largest=arr[i];
            }
            else if(arr[i]>s_largest && arr[i]<largest){
                t_largest=s_largest;
                s_largest=arr[i];
            }
            else if(arr[i]>t_largest && arr[i]<s_largest)t_largest=arr[i];
        }
        if(t_largest==LLONG_MIN)return largest;
        return t_largest;
    }
};

File: 0414-third-maximum-number/0414-third-maximum-number.java
class Solution {
    public int thirdMax(int[] arr) {
        Arrays.sort(arr);
        int mx1=arr[arr.length-1];
        int mx2 = mx1;
        for(int i=arr.length-1;i>=0;i--){
            if(arr[i]<mx1){
                mx2 = arr[i];
                break;
            }
        }
        for(int i=arr.length-1;i>=0;i--){
            if(arr[i]<mx2){
                return arr[i];
            }
        }
        return mx1;
    }
}